### PR TITLE
Save alias

### PR DESCRIPTION
**Pull request by 1ntEgr8**
This addresses #1731

I tried to implement `Display` for `Block` to reverse engineer the command from the struct, but that approach ended up being rather tricky and unnecessary. I also experimented with deserializing it using `serde`

I settled with adding a field called `raw_input` to `Context` that contains the exact command input by the user.

### CODE REVIEW

File: crates/nu-cli/src/commands/alias.rs [hunk 1]
```diff
@@ -59,19 +64,58 @@ impl WholeStreamCommand for Alias {
...
+        let mut raw_input = args.raw_input.clone();
+        let (AliasArgs { name, args: list, block, save}, ctx) = args.process(&registry).await?;
+        let mut processed_args: Vec<String> = vec![];
+
+        if let Some(true) = save {
+            let mut result = crate::data::config::read(name.clone().tag, &None)?;
+
+            // process the alias to remove the --save flag
+            let left_brace = raw_input.find('{').unwrap();
+            let right_brace = raw_input.rfind('}').unwrap();
```
Comment (jonathandturner) [inline crates/nu-cli/src/commands/alias.rs:82]: I'm surprised that clippy let you `.unwrap()` here. We went through a phase a while back to see if we could remove unwraps and use things that don't panic as easily.  You could use something like `if left Ok(v) = ...` instead

File: tmux-client-3904.log [hunk 2]
```diff
@@ -0,0 +1,42 @@
+1589828452.057215 client started (3904): version 2.9a, socket /private/tmp/tmux-501/default, protocol 8
```
Comment (jonathandturner) [inline tmux-client-3904.log:1]: This looks like you didn't mean to add it to the PR

**Comment by 1ntEgr8** [comment]
Noticed a bug in saving similar aliases. Instead of looking for the same alias name, it would compare the entire string.

**Comment by jonathandturner** [comment]
@1ntEgr8 - I think for now we can append to the end of the file, as that will replace any alias with the same name that came earlier.

**Comment by jonathandturner** [comment]
Oh I see what you've done. Yeah, that's fine. In the future, we can make it easier to search for matches, but this is good for now.

**Comment by jonathandturner** [comment]
Thanks!

**Pull request merged**